### PR TITLE
add e2e framework

### PR DESCRIPTION
**Pull request by Crazy-Kitty**
add part code for yurt e2e

### CODE REVIEW

**Comment by rambohe-ch** [comment]
lgtm

**Comment by Fei-Guo** [comment]
- Please sign the cla.
- Please add file header for all the new files if applicable. 
- Please explain why we need ecs.go and ens.go ? since you mentioned that the e2e test will be run against a minikube cluster.

File: hack/lib/build.sh [hunk 1]
```diff
@@ -76,10 +77,25 @@ build_binaries() {
...
+    for binary in "${statics[@]:+${statics[@]}}"; do
       echo "Building ${binary}"
       go build -o $(get_output_name $binary) -ldflags "${goldflags:-}" -gcflags "${gcflags:-}" ${goflags} $YURT_ROOT/${binary}
     done
+    for binary in "${tests[@]:+${tests[@]}}"; do
+      echo "Building ${binary}"
+      local testpkg="$(dirname ${binary})"
+      local filename="$(basename ${binary})"
+      go test -c  -gcflags "${gcflags:-}" ${goflags} -o $filename "$YURT_ROOT/${testpkg}"
+    done
```
Comment (charleszheng44) [inline hack/lib/build.sh:100]: could you create a function just for test? say `run_test()` and make the build logic looks like following:
`make e2e` -> `hack/make-rules/test.sh` -> `hack/lib/test.sh run_test()`

File: docs/tutorial/yurt-e2e-test.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Yurt-e2e-test 
+This tutorial shows how to build binary to \_output/bin/$GOOS/$GOARCH/yurt-e2e-test, such as: \_output/bin/darwin/amd64/yurt-e2e-test. And how to run yurt-e2e-test. We assume a minikube cluster  (version 1.14 or less) is installed and the cluser has been converted to Openyurt.
```
Comment (charleszheng44) [inline docs/tutorial/yurt-e2e-test.md:2]: This tutorial shows how to build and run the e2e test for openyurt.

File: docs/tutorial/yurt-e2e-test.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+## Build
+This tutorial demonstrates how to use `yurt-e2e-test` to test OpenYurt.
```
Comment (charleszheng44) [inline docs/tutorial/yurt-e2e-test.md:5]: First, let's build the e2e binary `yurt-e2e-test` by

File: docs/tutorial/yurt-e2e-test.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+1) First, enter openyurt work path:
```
Comment (charleszheng44) [inline docs/tutorial/yurt-e2e-test.md:6]: 1)  entering the openyurt work path:

File: docs/tutorial/yurt-e2e-test.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+```bash
+ $cd openyurt
+```
+
+2) Then, build test binary:
```
Comment (charleszheng44) [inline docs/tutorial/yurt-e2e-test.md:11]: 2) building the binary:

File: docs/tutorial/yurt-e2e-test.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+```bash
+$ make e2e
+```
+or as below:
+```
+$ make WHAT=test/e2e/yurt-e2e-test
+```
+
+## RUN Test
+1) So,you can use yurt-e2e-test binary to test openyurt.
```
Comment (charleszheng44) [inline docs/tutorial/yurt-e2e-test.md:21]: The functionality of running the test automatically is under development. For now, after the `yurt-e2e-test` has been successfully built,  we can run the test manually:

**Comment by Crazy-Kitty** [comment]
> * Please sign the cla.
> * Please add file header for all the new files if applicable.
> * Please explain why we need ecs.go and ens.go ? since you mentioned that the e2e test will be run against a minikube cluster.

> Thanks for your comment and I have signed the cla  and added file header.
> Openyurt can run on the node such aliyun-ecs and aliyun-ens, so e2e test will support node autonomy through aliyun-sdk to operate ecs and ens if user agree to do this. Minikube is an example to run yurt-e2e-test.

**Comment by Crazy-Kitty** [comment]
> * Please sign the cla.
> * Please add file header for all the new files if applicable.
> * Please explain why we need ecs.go and ens.go ? since you mentioned that the e2e test will be run against a minikube cluster.

> Thanks for your comment and I have signed the cla  and added file header.
> Openyurt can run on the node such aliyun-ecs and aliyun-ens, so e2e test will support node autonomy through aliyun-sdk to operate ecs and ens if user agree to do this. Minikube is an example to run yurt-e2e-test.

File: hack/lib/build.sh (hunk 1 above)
Comment (Crazy-Kitty) [reply hack/lib/build.sh]: thanks for you comment. I will do like this.

**Pull request merged**